android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ActionConfig.kt

```diff
@@ -39,13 +39,18 @@ data class ActionConfig(
 ) {
   fun paramsBundle(computedValuesMap: Map<String, Any> = emptyMap()): Bundle =
     Bundle().apply {
-      params.map { Pair(it.key, it.value.interpolate(computedValuesMap)) }.forEach {
-        putString(it.first, it.second)
-      }
+      params
+        .filter { it.paramType?.name.equals(PREPOPULATE_PARAM_TYPE) }
+        .map { Pair(it.key, it.value.interpolate(computedValuesMap)) }
```

Comment:
I think we can omit this `map` lambda and have the implementation in the `foreach`  below

Reply:
I've updated this to ommit the map